### ISSUE TITLE
Vfnet loss() missing 1 required positional argument: 'img_metas'

### ISSUE THREAD

**Issue by UygarUsta99** [airctic/icevision#1006]
## 🐛 Bug
**Describe the bug**
After the first epoch,when validating I get the error explained in title.
**To Reproduce**
 Training mmdet.vfnet model with backbone resnet50_fpn_mstrain_2x,image size 512.Batch size 16 and workers 8.Using CocoMetric,metric type COCOMetricType.bbox.15 epochs with the learning rate of 3e-4 and freeze_epochs = 1.

**Expected behavior**
First epoch was successful with no errors so expected whole training process to be done with no issues.

**Screenshots**
![ıcevision_vfnet_error](https://user-images.githubusercontent.com/82595202/146885952-f2446054-b276-4b30-8746-abaa6e7b1cb7.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04

**Additional context**

**Comment by FraPochetti** [comment]
Hi @UygarUsta99 can you please post this one in a #help channel on our [Discord](https://discord.gg/j5eMbjVT) server?
We are way more active there.

**Comment by UygarUsta99** [comment]
Will do as soon as possible.Thank you for responding so quickly.

**Comment by fstroth** [comment]
@UygarUsta99 does the problem still persist?

**Comment by UygarUsta99** [comment]
I have reproduced the getting started with object detection notebook on colab with vfnet,there were no errors.The error I have faced might be due to installation,I have to further experiment.I am closing this issue for now,thank you for helping.

**Comment by dnth** [comment]
For future reference, I encountered this error with `mmdet==2.20.0`. The error goes away with `mmdet==2.16.0`. I did not test other mmdet versions.

**Comment by UygarUsta99** [comment]
@dnth thank you for your feedback.

**Comment by loveisp** [comment]
it's because the if-else to self.training has changed: https://github.com/open-mmlab/mmdetection/commit/75238f307e7ea923e30b98cff198a14b9dfba485#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272

when validation step runs the number of return values decrease from 3 to 2, so L335 of base_dense_head.py the number of inputs of the func loss_inputs also decrease.

**Comment by dnth** [comment]
> it's because the if-else to self.training has changed: [open-mmlab/mmdetection@75238f3#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272](https://github.com/open-mmlab/mmdetection/commit/75238f307e7ea923e30b98cff198a14b9dfba485#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272)
> 
> when validation step runs the number of return values decrease from 3 to 2, so L335 of base_dense_head.py the number of inputs of the func loss_inputs also decrease.

I see.. is there a reason they make it that way? Or its a bug in the implementation?

**Comment by loveisp** [comment]
> 

I don't know why.. I just remove the if-else in vfnet_head.py and return the 3 values, then the notebook will run well. I don't know why someone add the if-else and make the validation or test step only return 2 values.

**Comment by loveisp** [comment]
> > it's because the if-else to self.training has changed: [open-mmlab/mmdetection@75238f3#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272](https://github.com/open-mmlab/mmdetection/commit/75238f307e7ea923e30b98cff198a14b9dfba485#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272)
> > when validation step runs the number of return values decrease from 3 to 2, so L335 of base_dense_head.py the number of inputs of the func loss_inputs also decrease.
> 
> I see.. is there a reason they make it that way? Or its a bug in the implementation?

I don't know why.. I just remove the if-else in vfnet_head.py and make the func return 3 values, then the notebook runs well. I don't know why someone add the if-else and make the validation or test step only return 2 values.

**Comment by dnth** [comment]
> > > it's because the if-else to self.training has changed: [open-mmlab/mmdetection@75238f3#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272](https://github.com/open-mmlab/mmdetection/commit/75238f307e7ea923e30b98cff198a14b9dfba485#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272)
> > > when validation step runs the number of return values decrease from 3 to 2, so L335 of base_dense_head.py the number of inputs of the func loss_inputs also decrease.
> > 
> > 
> > I see.. is there a reason they make it that way? Or its a bug in the implementation?
> 
> I don't know why.. I just remove the if-else in vfnet_head.py and make the func return 3 values, then the notebook runs well. I don't know why someone add the if-else and make the validation or test step only return 2 values.

I see.. thank you for reporting it here! I will take a look later and if its a bug, I will make PR for a fix. Thanks again!

**Comment by dnth** [comment]
I tried training vfnet with the mmdetection training script. It seems like the error does not occur

**Comment by dnth** [comment]
> it's because the if-else to self.training has changed: [open-mmlab/mmdetection@75238f3#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272](https://github.com/open-mmlab/mmdetection/commit/75238f307e7ea923e30b98cff198a14b9dfba485#diff-533f80fd026c64e139df3be8c5f6b0a5dac3c301350186b60f08c39d3354de01L272)
> 
> when validation step runs the number of return values decrease from 3 to 2, so L335 of base_dense_head.py the number of inputs of the func loss_inputs also decrease.

@ai-fast-track is this something we can fix in IceVision?
